src/messaging/ExchangeMgr.h

```diff
@@ -249,9 +249,13 @@ class DLL_EXPORT ExchangeManager : public SecureSessionMgrDelegate, public Trans
                            SecureSessionMgr * msgLayer) override;
 
     void OnNewConnection(SecureSessionHandle session, SecureSessionMgr * mgr) override;
+#if CHIP_CONFIG_TEST
+public: // Allow OnConnectionExpired to be called directly from tests.
+#endif  // CHIP_CONFIG_TEST
     void OnConnectionExpired(SecureSessionHandle session, SecureSessionMgr * mgr) override;
 
     // TransportMgrDelegate interface for rendezvous sessions
+private:
```

Comment:
We can save 'private' if we move OnMessageReceived before  OnConnectionExpired

Reply:
We can, but then we run the danger of someone adding something that ends up public in tests if they just put it at te end of the file.  So this is safer, imo.